src/frontend-samples/read-settings-sample/ExternalSettingsUI.tsx

```diff
@@ -0,0 +1,169 @@
+    this.saveSettings = this.saveSettings.bind(this);
+  }
+
+  private onIModelReady = (imodel: IModelConnection) => {
+    this.setState({ imodel });
+    AuthorizedFrontendRequestContext.create().then(async context => {
+      const imodelInfo = await this.getIModelInfo(this.props.iModelName, context);
+      this.setState({ projectId: imodelInfo.projectId, imodelId: imodelInfo.imodelId });
+    });
+  }
+
+  // This method serves to query projectId of project where iModel is stored in this example, 
+  // however in real application you might have it upfront
+  private async getIModelInfo(iModelName: string, context: AuthorizedFrontendRequestContext): Promise<{ projectId: string, imodelId: string }> {
+    // In testdrive the projectName matches iModelName.  That's not true in general.
+    const projectName = iModelName;
+    const connectClient = new ContextRegistryClient();
+    let project: Project;
+
+    try {
+      project = await connectClient.getProject(context, { $filter: `Name+eq+'${iModelName}'` });
+    } catch (e) {
+      throw new Error(`Project with name "${projectName}" does not exist`);
+    }
+
+    const imodelQuery = new IModelQuery();
+    imodelQuery.byName(iModelName);
+    const imodels = await IModelApp.iModelClient.iModels.get(context, project.wsgId, imodelQuery);
+    if (imodels.length === 0)
+      throw new Error(`iModel with name "${iModelName}" does not exist in project "${projectName}"`);
+
+    return { projectId: project.wsgId, imodelId: imodels[0].wsgId };
+  }
+
+  // Handler to read external settings when name in dropdown changes
+  handleSettingsChange(event: ChangeEvent<HTMLSelectElement>) {
+    const settingKey = event.target.value;
+    this.setState({ settingKey: settingKey });
+
+    AuthorizedFrontendRequestContext.create().then(async context => {
+      const response = await IModelApp.settings.getSetting(context, namespace, settingKey, true, this.state.projectId, this.state.imodelId);
+      let value;
+
+      switch (settingKey) {
+        case 'Json_Data':
+          value = JSON.stringify(response.setting || '', null, '   ');
+          break;
+        default:
+          value = response.setting || ''
+      }
+
+      this.setState({
+        settingResult: response,
+        settingValue: value
+      });
+    });
+  }
+
+  // Handler to get settings value into state, when you modify textarea element in the dialog
+  handleSettingsValueChange(event: any) {
+    const settingValue = event.target.value;
+    this.setState({ settingValue: settingValue });
+  }
+
+  // The showcase does not have permission to write data, it is expected to fail with 403 Forbidden. 
+  // However saveSetting method will work in your project with signed-in user, who has required permissions in the project.
+  saveSettings() {
+    this.setState({ saveInProgress: true });
+    AuthorizedFrontendRequestContext.create().then(async context => {
```

Comment:
The code that calls saveSetting should be triggered in the componentDidUpdate method.  Also, the implementation should be in a static method in the App.tsx file.

Reply:
made static and moved to app.tsx